pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageIdImpl.java

```diff
@@ -140,4 +151,16 @@ public BatchMessageAcker getAcker() {
         return acker;
     }
 
+    static int messageIdCompare(
+            long ledgerId1, long entryId1, int partitionIndex1, int batchIndex1,
+            long ledgerId2, long entryId2, int partitionIndex2, int batchIndex2
+    ) {
+        return ComparisonChain.start()
+                .compare(ledgerId1, ledgerId2)
```

Comment:
This code will be exd uted in hot paths probably.
Is it better to use raw java code? We could save method calls and allocations

Reply:
goog point! I think we can use raw java code